### PR TITLE
Reworks zombie revival

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
balance: (Real) Zombies no longer die from normal means. They will instead go into crit indefinitely and regenerate until they get back up again. To kill a zombie permanently you either need to remove the tumor, cut off their head, gib them, or use more elaborate methods like a wand of death or driving them to suicide.
balance: Slowed down zombie regeneration overall, since putting them in crit now no longer disables them for a full minute in average.
/:cl:

Why this change? Mostly to remove the fully_heal proc in the revive, allowing zombies to retain miscellaneous injuries and traumas they had in life, for consistency.
Plus you can now get crawling zombies with soft crit, and revival is a gradual, observable process instead of going from 0 to 100 in an instant. Realism!

Regen speed might be subject to change, especially if it turns out to be too easy to accumulate damage on a critted zombie in order to keep them down forever. Alternatively, mauling them so hard you reach extremely high levels of damage could instead kill them outright, since they are still flesh and bone after all.

Note to self: remember to let head cutting happen in crit in the limb rework.

### CODE REVIEW

**Comment by Shadowflame909** [comment]
The only downside I can see from this is dizzyness. Lings deal with it, people with healing viruses deal with it. The dizzy mechanic is such a shitty one and makes playing ss13 a pain.

**Comment by XDTM** [comment]
What even gives dizziness, aside from diseases

**Comment by Ratvar** [comment]
Get hit on the head for braindamage, dizzyness, knockdown

**Comment by ExcessiveUseOfCobblestone** [comment]
If you use realism while making a PR about a fictitious monster again i'm going to MOO you.

**Comment by GuyonBroadway** [comment]
You can also spam a flashbulbs AoE attack to make a room full of people dizzy and robotics can mass fabricate that shit.

File: code/modules/mob/living/carbon/human/species_types/zombies.dm [hunk 1]
```diff
@@ -46,15 +47,25 @@
...
-		// Zombies only move around when not in crit, they instantly
-		// succumb otherwise, and will standup again soon
+		
+//Congrats you somehow died so hard you stopped being a zombie
+/datum/species/zombie/infectious/spec_death(mob/living/carbon/C)
+	. = ..()
+	var/obj/item/organ/zombie_infection/infection
+	infection = C.getorganslot(ORGAN_SLOT_ZOMBIE)
+	if(infection)
+		qdel(infection)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/zombies.dm]: QDEL_NULL probably

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ishitbyabullet** [comment]
if zombies don't have a big damage multipler to their head then they should. it just makes sense.

**Comment by XDTM** [comment]
If zombies had it, it would make sense for everyone tbh, but zombies do have a head weakness since cutting it off will perma kill them

File: code/modules/mob/living/carbon/human/species_types/zombies.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/human/species_types/zombies.dm]: it's a local var tho, does it make sense?

Comment (Cyberboss) [reply code/modules/mob/living/carbon/human/species_types/zombies.dm]: nvm actually

**Comment by kevinz000** [comment]
project zomboid

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**